### PR TITLE
Update stm32 unittest instructions

### PR DESCRIPTION
**Pull request by donmr**

### CODE REVIEW

**Comment by drowe67** [comment]
@donmr Compiled everything very nicely, however Step 2/ in README_unittest.txt stopped after a few seconds:
```
$ ./scripts/run_stm32_tst tst_ofdm_demod quick
```
I added a script to kills all the child processes in the test, as gdb doesn't respond to Ctrl-C.  Pls let me know if there is a better way to neatly stop tests that have hung.

**Comment by donmr** [comment]
It needs "--load" at the end.  I will figure out how to cover that in the 
README.

File: stm32/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  requires a registration on the STM website.  Save the zip file
+  somewhere safe and then extract it anywhere you like. You will have
+  to tell cmake where the unzipped library is by giving the variable
+  PERIPHLIBDIR the location of top level directory, e.g. for version
+  1.8.0 this is STM32F4xx_DSP_StdPeriph_Lib_V1.8.0.
+
+  In this example we will assume the library has been unzipped in ~/Downloads.
+
+5. Configure the build system by running cmake and pointing to the svn
+   checkout of the codec2-dev/stm32 directory and the StdPeriph lib:
```
Comment (db4ple) [inline stm32/README.md]: we should change that to git related instructions

File: stm32/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Building for the stm32
+
+## Quickstart
+
+1. Build codec2 for your host system, see codec2/README.md and codec2/INSTALL
+
+2. Install agcc-arm-none-eabi toolchain:
```
Comment (db4ple) [inline stm32/README.md]: We should mention NOT to use the distributions packages as they are often outdate (or even have serious issues such as the one in Ubuntu 18.04).

File: stm32/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   ```
+   
+## Directories
+
+Directory | Notes 
+---|---
+cmake | cmake support files for the stm32
+inc | top level sm1000 source, drivers, and some legacy test code
+src | top level sm1000 source, drivers, and some legacy test code
+unitest | comprehensive set of automated unit tests for the stm32 700D port
```
Comment (db4ple) [inline stm32/README.md]: unitest -> unittest

**Comment by drowe67** [comment]
1. > It needs "--load" at the end. I will figure out how to cover that in the README.
```
./run_stm32_tst tst_ofdm_demod quick --load
<snip>
Breakpoint 1, Reset_Handler () at /home/david/codec2/stm32/unittest/src/startup_stm32f4xx.s:119
119	  bl  .
2019-04-14T06:39:49 ERROR gdb-server.c: cannot recv: -2

Test tst_ofdm_demod_quick FAILED!
```
2.  When I tried:
```
./run_all_stm32_tests 
tst_codec2_enc_1600
```
I seemed to get stuck.  No output for several minutes.  But perhaps running tests this way is now out if date, given https://github.com/drowe67/codec2/pull/24.

**Comment by db4ple** [comment]
Yes, the test could not have finished as there was no 1600 support in the binary. We fixed that in the new PR.

**Comment by drowe67** [comment]
> It needs "--load" at the end. I will figure out how to cover that in the README.
> […](#)
> On April 12, 2019 21:50:47 drowe67 ***@***.***> wrote: @donmr Compiled everything very nicely, however Step 2/ in README_unittest.txt stopped after a few seconds: ``` $ ./scripts/run_stm32_tst tst_ofdm_demod quick ``` I added a script to kills all the child processes in the test, as gdb doesn't respond to Ctrl-C. Pls let me know if there is a better way to neatly stop tests that have hung. -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [#22 (comment)](https://github.com/drowe67/codec2/pull/22#issuecomment-482776087)

@donmr has this been addressed now?

**Comment by donmr** [comment]
I do not see the updated stm32/unittest/README_unittest.txt in master.  I put in more instructions for running the tests and fixed a few errors.  That was what this pull request was for.

**Comment by drowe67** [comment]
> I do not see the updated stm32/unittest/README_unittest.txt in master. I put in more instructions for running the tests and fixed a few errors. That was what this pull request was for.

That's because it hasn't been merged.  I have also updated codec2/stm32/README.md in this PR.  Don, if you can pls address the [issue above](https://github.com/drowe67/codec2/pull/22#issuecomment-482776242) I think we are set to merge and close this PR.

**Comment by drowe67** [comment]
@donmr I just tried that line, is this the correct output?
```
david@bear:~/codec2/stm32/unittest/scripts$ git branch 
* issue-20
david@bear:~/codec2/stm32/unittest/scripts$ ./run_stm32_tst tst_ofdm_demod quick --load
<snip>
Transfer rate: 12 KB/sec, 9155 bytes/write.
2019-04-16T17:29:11 INFO gdb-server.c: Found 6 hw breakpoint registers
Breakpoint 1 at 0x80126b4: file /home/david/codec2/stm32/unittest/src/startup_stm32f4xx.s, line 119.
Breakpoint 2 at 0x80148d0
Note: automatically using hardware breakpoints for read-only addresses.
semihosting: unsupported call 0xc
semihosting: unsupported call 0x9
semihosting: unsupported call 0xc
semihosting: unsupported call 0x9
semihosting: unsupported call 0xc
semihosting: unsupported call 0x9

Breakpoint 1, Reset_Handler ()
    at /home/david/codec2/stm32/unittest/src/startup_stm32f4xx.s:119
119	  bl  .
2019-04-16T17:29:56 ERROR gdb-server.c: cannot recv: -2

Test tst_ofdm_demod_quick PASSED
```

Is this was expected output (given the ERROR message above)?

**Comment by donmr** [comment]
I have been seeing that too, st-util is disconnecting.  Can we ignore it as we switch to openocd?

**Comment by donmr** [comment]
What type of board do you have atached to bear, an older discovery with stlink-v1 or a newer one with stlink-v2??  I see a difference in the ocd_stderr.log files:

On bear:
    adapter speed: 2000 kHz
On reid1:
   Info : Unable to match requested speed 2000 kHz, using 1800 kHz
   Info : Unable to match requested speed 2000 kHz, using 1800 kHz
   adapter speed: 1800 kHz

It procededs to be "unable" to provide several other speeds later.

**Comment by db4ple** [comment]
@donmr : I don't know which board is connected to bear, but I have a stm32f429 discovery connected (stlink v2) and it does the same as your board, i.e. 
```
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
adapter speed: 1800 kHz
```

**Comment by drowe67** [comment]
> What type of board do you have atached to bear, an older discovery with stlink-v1 or a newer one with stlink-v2?? I see a difference in the ocd_stderr.log files:

The board is a 5 year old Discovery - how can I determine if it is v1 or v2?

**Comment by db4ple** [comment]
Don't worry, you have v2. OpenOCD tells you this:
Info : STLINK V2J14S0 (API v2) VID:PID 0483:3748
However, you have a very old firmware  for the stlink, currently there is V2J33S0 available. I had some issue with very old stlink firmware version, but it seems to not a problem with your board.
Anyway, I don't think/hope that the issues of Don are related to the stlinks firmware, it is worth a try to upgrade to the newest. ST provides a Java tool as part of the normal firmware update tool which works on Linux (and Windows and MacOS X).

**Comment by drowe67** [comment]
> I have been seeing that too, st-util is disconnecting. Can we ignore it as we switch to openocd?

We need to make sure the examples work, but sure, that can be done as part of more edits to README_unittest to reflect the use of openocd in another PR.

**Comment by drowe67** [comment]
> Don't worry, you have v2. OpenOCD tells you this:
> Info : STLINK V2J14S0 (API v2) VID:PID 0483:3748
> However, you have a very old firmware for the stlink, currently there is V2J33S0 available. I had some issue with very old stlink firmware version, but it seems to not a problem with your board.
> Anyway, I don't think/hope that the issues of Don are related to the stlinks firmware, it is worth a try to upgrade to the newest. ST provides a Java tool as part of the normal firmware update tool which works on Linux (and Windows and MacOS X).

OK

**Comment by drowe67** [comment]
> Don't worry, you have v2. OpenOCD tells you this:
> Info : STLINK V2J14S0 (API v2) VID:PID 0483:3748
> However, you have a very old firmware for the stlink, currently there is V2J33S0 available. I had some issue with very old stlink firmware version, but it seems to not a problem with your board.
> Anyway, I don't think/hope that the issues of Don are related to the stlinks firmware, it is worth a try to upgrade to the newest. ST provides a Java tool as part of the normal firmware update tool which works on Linux (and Windows and MacOS X).

OK I'll try to upgrade this weekend.  Happy to buy a couple of new boards if that helps too.

**Comment by donmr** [comment]
The new boards have "STM32F407-DISC1" printed by the USB connector, the
original ones do not.

They also have different USB IDs, yours has 0483:3748, my new one has 0483:374b.

**Comment by donmr** [comment]
My board has the newest as of a few months ago.

Info : STLINK V2J30M21 (API v2) VID:PID 0483:374B

Is there a newer one??

**Comment by donmr** [comment]
A new out of the box board has:

Info : STLINK V2J25M14 (API v2) VID:PID 0483:374B

(I bought 2 last time)

**Pull request merged**